src/http.zig

```diff
@@ -747,6 +763,10 @@ pub const HTTPThread = struct {
     }
 
     pub fn connect(this: *@This(), client: *HTTPClient, comptime is_ssl: bool) !NewHTTPContext(is_ssl).HTTPSocket {
+        if (client.unix_socket_path.len > 0) {
+            return try this.context(is_ssl).connectSocket(client, client.unix_socket_path);
+        }
+
         if (client.http_proxy) |url| {
```

Comment:
should we error on Bun__fetch if we try to use http_proxy + unix socket?

Reply:
Yeah we probably should

Reply:
added

Reply:
we also need a test for this case